5663/src/cpp/Robot.cpp

```diff
@@ -1,25 +1,109 @@
 #include "WPILib.h"
-
 #include "curtinfrc/math.h"
-
 #include <iostream>
+#include <string>
+#include <SmartDashboard/SmartDashboard.h>
+#include <PowerDistributionPanel.h>
+#include "ctre/Phoenix.h"
 
 using namespace frc;
 using namespace curtinfrc;
+using namespace std;
 
 class Robot : public IterativeRobot {
-    void RobotInit() {
-        std::cout << "0.5^2 = " << math::square_keep_sign(0.5) << std::endl;
-    }
-
-    void AutonomousInit() { }
-    void AutonomousPeriodic() { }
-    
-    void TeleopInit() { }
-    void TeleopPeriodic() { }
-
-    void TestInit() { }
-    void TestPeriodic() { }
+  XboxController *xbox = new XboxController(0);
+  PowerDistributionPanel *pdp = new PowerDistributionPanel(0);
+  SendableChooser<int*> *AutoChooser = new SendableChooser<int*>; // Choose auto mode
+  TalonSRX *left1 = new TalonSRX(1), *left2 = new TalonSRX(2), *left3 = new TalonSRX(3),
+    *right1 = new TalonSRX(4), *right2 = new TalonSRX(5), *right3 = new TalonSRX(6);
+public:
+  // Configuration settings: (maybe make a sperate file for these?)
+  double deadzone = 0.04; //Stop the robot being a sneaky snail
+  // Regular variables
+  string gameData;
+  bool A,B,X,Y,LB,RB,back,start,LS,RS;
+  double LX,LY,RX,RY,LT,RT,Dpad;
+  int Auto;
+  void RobotInit() {
+    std::cout << "0.5^2 = " << math::square_keep_sign(0.5) << std::endl; //?
+    AutoChooser->AddDefault("Cross Baseline",(int*) 0);
+    AutoChooser->AddObject("Auto 1",(int*) 1);
+    AutoChooser->AddObject("Auto 2",(int*) 2);
+    getValues();
+    updateDash();
+  }
+
+  void Drive(double l, double r) {  //probably a wpi class to do this for us but I couldnt find one so.
+    if(-deadzone < l && l < deadzone) l == 0;
+    if(-deadzone < r && r < deadzone) r == 0;
+    l *= abs(l);
+    r *= abs(r); // square inputs
+    left1->Set(ControlMode::PercentOutput,l); left2->Set(ControlMode::PercentOutput,l); left3->Set(ControlMode::PercentOutput,l);
+    right1->Set(ControlMode::PercentOutput,l); right2->Set(ControlMode::PercentOutput,l); right2->Set(ControlMode::PercentOutput,l);
+  }
+
+  void getValues() { //Set variables for the xbox controller values for easy coding (put in seperate file?)
+    A = xbox->GetRawButton(1);
+    B = xbox->GetRawButton(2);
+    X = xbox->GetRawButton(3);
+    Y = xbox->GetRawButton(4);
+    LB = xbox->GetRawButton(5);
+    RB = xbox->GetRawButton(6);
+    back = xbox->GetRawButton(7);
+    start = xbox->GetRawButton(8);
+    LS = xbox->GetRawButton(9);
+    RS = xbox->GetRawButton(10);
+    LX = xbox->GetRawAxis(0);
+    LY = xbox->GetRawAxis(1);
+    LT = xbox->GetRawAxis(2);
+    RT = xbox->GetRawAxis(3);
+    RX = xbox->GetRawAxis(4);
+    RY = xbox->GetRawAxis(5);
+    Dpad = xbox->GetPOV();
```

Comment:
All these values are members of the XboxController class `(get<>Button, get<X,Y>Axis, getTriggerAxis)`

http://first.wpi.edu/FRC/roborio/release/docs/cpp/classfrc_1_1XboxController.htm

Reply:
I was thinking just less to write but ok

Reply:
also whats the point? the variables are clearly named and that stuff is behind the scenes

Reply:
The idea is to not save these variables in the class state, but instead access the XboxController object directly. If you save these in the class state, you're opening up the possibility to access out-of-date variables which can breed bugs. It also doesn't allow you to add a second controller in the future